lib/category_item.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import 'package:meals_app/category_meals_screen.dart';
+
+class CategoryItem extends StatelessWidget {
+  final String id;
+  final String title;
+  final Color color;
+
+  const CategoryItem(
+      {super.key, required this.title, required this.color, required this.id});
+
+  void selectCategory(BuildContext ctx) {
+    Navigator.of(ctx).pushNamed(
+        CategoryMealsScreen.routeName, arguments: {'id': id, 'title': title,}
```

Comment:
I would rather create a small class to pass the arguments. That way we can cast the incoming arguments to that class on the receiving end too, creating a bit more type safety